platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -123,17 +123,24 @@ public Object saveInstance(EntityRecord entityRecord) {
     @Override
     @Transactional
     public Object saveInstance(EntityRecord entityRecord, Long deleteValueFieldId) {
-        EntityDto entity = getEntity(entityRecord.getEntitySchemaId());
-        validateCredentials(entity);
-        validateNonEditableProperty(entity);
+        EntityDto[] entityTarget = new EntityDto[]{null};
```

Comment:
avoid arrays, prefer collections. Moreover, why do we initiate it with a null value?